test/client/docs_test.rb

```diff
@@ -13,19 +13,24 @@
           :doc1 => {
             :_source => { :enabled => true }, :_all => { :enabled => false },
             :properties => {
-              :title  => { :type => "string", :analyzer => "standard", :term_vector => "with_positions_offsets" },
-              :author => { :type => "string", :index => "not_analyzed" }
+              :title  => { :type => "text", :analyzer => "standard", :term_vector => "with_positions_offsets" },
+              :author => { :type => "keyword", :index => "not_analyzed" }
             }
           },
           :doc2 => {
             :_source => { :enabled => true }, :_all => { :enabled => false },
             :properties => {
-              :title  => { :type => "string", :analyzer => "standard", :term_vector => "with_positions_offsets" },
-              :author => { :type => "string", :index => "not_analyzed" }
+              :title  => { :type => "text", :analyzer => "standard", :term_vector => "with_positions_offsets" },
+              :author => { :type => "keyword", :index => "not_analyzed" }
             }
           }
         }
 
+      # COMPATIBILITY
+      if es_version_5_x?
+        @index.update_mapping("percolator", { :properties => { :query => { :type => "percolator"}}})
+      end
```

Comment:
Same comment here about creating a `mappings` hash and then calling `@index.create`.